Sphinx ^= 6.1.0 couldn't build our documentation

**Description**
In the latest version of Sphinx (from [6.1.0](https://www.sphinx-doc.org/en/master/changes.html#release-6-1-0-released-jan-05-2023) to [6.1.3.](https://www.sphinx-doc.org/en/master/changes.html#release-6-1-3-released-jan-10-2023)), there is a breaking change that prevent the Macaron's documentation from being built successfully.
When building the documentation with `make clean docs`, there following error messages is seen (some part of the message is removed for visibility):
```
Running Sphinx v6.1.0
making output directory... done
[autosummary] generating autosummary for: "***"
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 18 source files that are out of date
updating environment: [new config] 18 added, 0 changed, 0 removed
ERROR: [numpydoc] While processing docstring for 'macaron.code_analyzer.call_graph.Node'                                                                                                            

Extension error (numpydoc.numpydoc):
Handler <function mangle_docstrings at 0x7fbe18f79d00> for event 'autodoc-process-docstring' threw an exception (exception: Expected a class or None, but got ~Node)
make[1]: *** [Makefile:23: html] Error 2
make[1]: Leaving directory '***'
make: *** [Makefile:230: docs/_build/html/index.html] Error 2
```
The message indicates that the error happened for the global variable [here](https://github.com/oracle-samples/macaron/blob/babd2e784fcea81ed6b383ad3ad992f9d6ab8cc7/src/macaron/code_analyzer/call_graph.py#L9).

**Solution**
- We could revert the version of Sphinx back to 6.0.0 so that the error won't happen.

--- a/src/macaron/code_analyzer/call_graph.py
+++ b/src/macaron/code_analyzer/call_graph.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2022 - 2022, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2022 - 2023, Oracle and/or its affiliates. All rights reserved.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
 
 """This module contains classes to generate build call graphs for the target repository."""
@@ -7,11 +7,14 @@ from collections.abc import Iterable
 from typing import Generic, TypeVar
 
 Node = TypeVar("Node", bound="BaseNode")
-"""This binds type ``Node`` to ``BaseNode`` and any of its subclasses.
+# The documentation below for `TypeVar` is commented out due to a breaking
+# change in Sphinx version (^=6.1.0).
+# Reported at: https://github.com/oracle-samples/macaron/issues/58.
+# """This binds type ``Node`` to ``BaseNode`` and any of its subclasses.
 
-Therefore, any node of type ``Node`` that is stored in the call graph
-container will be a subtype of ``BaseNode``.
-"""
+# Therefore, any node of type ``Node`` that is stored in the call graph
+# container will be a subtype of ``BaseNode``.
+# """
 
 
 class BaseNode(Generic[Node]):

--- a/src/macaron/output_reporter/results.py
+++ b/src/macaron/output_reporter/results.py
@@ -62,7 +62,10 @@ class DepSummary(TypedDict):
 
 
 RecordNode = TypeVar("RecordNode", bound="Record")
-"""This binds type ``RecordNode`` to ``Record`` and any of its subclasses."""
+# The documentation below for `TypeVar` is commented out due to a breaking
+# change in Sphinx version (^=6.1.0).
+# Reported at: https://github.com/oracle-samples/macaron/issues/58.
+# """This binds type ``RecordNode`` to ``Record`` and any of its subclasses."""
 
 
 @dataclass

